### PR TITLE
Flag tags shown in different colours in the Flagr UI

### PR DESCRIPTION
**Pull request by SaxJ**
## Description

Tag value is hashed, and the tag UI elements use that hash value for the colour. The hash value is the Hue in a css `HSLA` colour, which is applied to the tag UI elements. A HSLA colour is used, as it is easier to ensure the colour is readable. 

## Motivation and Context

https://github.com/checkr/flagr/issues/438
A small quality of life improvement. Different tags colours allow a user to more easily see what flags have what tags when scanning through the tag list. 
Screenshot_2021-04-19 Flagr
Screenshot_2021-04-19 Flagr(1)

## How Has This Been Tested?

Flagr docker image was run, and UI served using `npm run serve` pointing at the local Flagr docker instance. 
1. Homepage UI visually inspected
2. Created Flag, UI with no tags present is the same
3. Created tags of varying random strings. Checked visually for legibility. Adding and removing tags still works.
4. Homepage UI shows tags correctly with the same colours.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] UI only change.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: browser/flagr-ui/src/App.vue [hunk 1]
```diff
@@ -39,16 +39,17 @@ const version = require("../package.json").version || "1.0.0";
 export default {
   name: "app",
   data: () => ({
-    version,
-  }),
+    version
```
Comment (marceloboeira) [inline browser/flagr-ui/src/App.vue:42]: I would leave the commas at the end since it avoids conflicts on future changes for lines that are unrelated...

File: browser/flagr-ui/src/helpers/helpers.js [hunk 2]
```diff
@@ -40,10 +40,27 @@ function handleErr (err) {
...
+}
+
 export default {
   indexBy,
   pluck,
   sum,
   get,
-  handleErr
-}
+  handleErr,
```
Comment (marceloboeira) [inline browser/flagr-ui/src/helpers/helpers.js:64]: same here, leave the commas at the end to avoid changing unrelated lines and creating conflicts

File: browser/flagr-ui/src/helpers/helpers.js [hunk 3]
```diff
@@ -40,10 +40,27 @@ function handleErr (err) {
...
+    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    hash = hash & hash;
+  }
+  return hash % 360;
+}
+
+function stringToColour(str, lightness = 90) {
+  let hue = getHue(str);
+
+  return `hsla(${hue}, 100%, ${lightness}%, 1)`;
```
Comment (marceloboeira) [inline browser/flagr-ui/src/helpers/helpers.js:56]: could we simplify it? maybe
```javascript 
return `hsla(${getHue(str)}, 100%, ${lightness}%, 1)`;
```

File: browser/flagr-ui/src/helpers/helpers.js [hunk 4]
```diff
@@ -40,10 +40,27 @@ function handleErr (err) {
   }
 }
 
+function getHue(str) {
```
Comment (marceloboeira) [inline browser/flagr-ui/src/helpers/helpers.js:43]: do we test this somewhere?

**Review by marceloboeira — CHANGES_REQUESTED** [review]
Could we test the feature and add some intermediate tests for the hashing function?

File: browser/flagr-ui/src/helpers/helpers.js (hunk 4 above)
Comment (SaxJ) [reply browser/flagr-ui/src/helpers/helpers.js:43]: Apologies for my slow responses.

I can't see any front-end testing set up or any tests for the existing javascript. Quite possible I'm missing something though.
I'm happy to set up a testing framework for the front-endif there are none. In this PR or another. :)

Comment (marceloboeira) [reply browser/flagr-ui/src/helpers/helpers.js:43]: no worries! I'm not much aware of the FE, that's something @zhouzhuojie would know better

Comment (zhouzhuojie) [reply browser/flagr-ui/src/helpers/helpers.js:43]: Yup, frontend testing is light - may need to setup something first. I can investigate getting something simple like a small js test to start the CI workflow for frontend testing.
